src/bp/nlu-core/entities/custom-entity-extractor.ts

```diff
@@ -162,17 +180,39 @@ export const extractListEntities = (
   utterance: Utterance,
   list_entities: ListEntityModel[]
 ): EntityExtractionResult[] => {
-  const cacheKey = utterance.toString({ lowerCase: true })
+  return _.flatMap(_extractListEntities(utterance, list_entities), ({ extractions }) => extractions)
```

Comment:
This is a bit hard to read but that might be my crush for functional speaking

suggestion : 
```ts
return _.chain(list_entities)
 .map(extractForListModel.bind(null, utterance))
 .flatMap('extractions')
 .value()
```

Note that you don't need to filter on extraction.length it'll be done implicitly. flatMap (reduce + concat) will concat resulting array with an empty array

Reply:
:+1: